### PR TITLE
Switch form Yarn to `npm`

### PR DESCRIPTION
**Pull request by renchap**
This PR is based on #27073

Yarn 1 is no longer maintained and suffers from many bugs (including https://github.com/eslint/eslint/discussions/17215#discussioncomment-6609097 which prevents multiple package upgrades)

Our setup should now properly support corepack, so installing the package manager is as simple as corepack prepare (and maybe corepack enable before, if it has never been ran on this machine and is not down when installing node). It will automatically pick the correct `npm` version, and warn the user if they try to use another package manager (like `yarn`).

I needed to:
- monkey patch Webpacker to call `npm` instead of `yarn`
- add additional dependencies, and force version for some peer-dependencies that are no longer updated

### CODE REVIEW

**Comment by renchap** [comment]
Alternative PR to switch to `pnpm`: #27299
Alternative PR to switch to Yarn 4: #27073

My preference goes to Yarn 4, because it has better performance, supports local patches, and keep the familiar `yarn` command that is everywhere in the docs.

But this is a working alternative if we want to get back to plain `npm`.

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -15,6 +15,6 @@ RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
 RUN gem install foreman
 
 # [Optional] Uncomment this line to install global node packages.
-RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && npm install -g yarn" 2>&1
+RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && corepack enable" 2>&1
```
Comment (nschonni) [inline .devcontainer/Dockerfile:18]: corepack shouldn't be needed if going to NPM

File: Dockerfile [hunk 2]
```diff
@@ -13,7 +13,7 @@ ENV DEBIAN_FRONTEND="noninteractive" \
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 WORKDIR /opt/mastodon
-COPY Gemfile* package.json yarn.lock /opt/mastodon/
+COPY Gemfile* package.json package-lock.json /opt/mastodon/
```
Comment (nschonni) [inline Dockerfile:16]:
```suggestion
COPY Gemfile* package*.json /opt/mastodon/
```

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (renchap) [reply .devcontainer/Dockerfile:18]: It depends if we want to lock the `npm` version. I am not sure what is the best practice here.

**Comment by renchap** [comment]
Closing as https://github.com/mastodon/mastodon/pull/27073 has been merged.

**Pull request closed without merging**